DataDog/datadog-agent
# AMLII-1701 Enabling stream-logs in a flare triggered by remote config

### What does this PR do?

This PR integrates the streamlogs functionality into the remote configuration for debug flare option. It allows support team to get visibility on their log collection process to troubleshoot missing logs using remote config. 

### Motivation

The motivation behind this change is to improve visibility and debugging capabilities for troubleshooting missing logs process. By streaming logs directly from the remote configuration client, engineers and supports can gain immediate insights into the configuration process and quickly identify and resolve issues.

### Possible Drawbacks / Trade-offs

One possible drawback of this change is the potential increase in network traffic or flare creation time due to the streaming of logs (by default it adds additional 60s). This could impact performance, especially in environments with limited network resources. However, the benefits of improved visibility and debugging capabilities are likely to outweigh this potential drawback.

### Describe how to test/QA your changes

To test these changes, follow these steps:
1. Adjust the datadog.yaml file to enable the log agent and then enable stream-logs:
```
logs_enabled: true
streaming: 
    enable_streamlogs: true
```

2. Restart the agent and check if the config is applied by running:
```
datadog-agent config get enable_streamlogs
```

As of right now since rc args are not yet available but will be by rc team, I've been testing by instead of using RC args logic, i put the streamlogs logic into the debug level mode from fleet automation. Once the logic runs cleanly and the feature is functional, i put the logic of streamlogs back behind the rc arg. 

### Possible Drawbacks / Trade-offs
Gathering streamlogs information in the flare can increase generation times. By default it is 60 seconds + network delays
